src/blog/jwt-identification-and-hot-streams/index.md

```diff
@@ -0,0 +1,173 @@
+  <source src="./connect-benchmark.h264.mp4" ttype="video/mp4; codecs=avc1.4D401E,mp4a.40.2">
+</video>
+</figure>
+
+It's more than 2x faster now! And even more importantly, we've reduced the stress on the RPC server. This could be especially useful during _connection avalanches_.
+
+> JWT identification allows you to standardize the authentication flow for WebSockets, protects from cross-site WebSocket hijacking and results in a performance boost!
+
+Luckily, you don't need to write any of the code above yourself, even the token generation part is done for you via a little companion gem called [anycable-rails-jwt][].
+
+The only question left: how can we gracefully handle token expiration? AnyCable Go uses a specific disconnect message to distinguish expiration from unauthorized access (sends a `{"type":"disconnect","reason":"token_expired"}` message). You can use it to refresh the token. Again, we're glad to provide an out-of-the-box solution for this in our brand new [anycable-client-refresh-tokens][]!
+
+## "Hot-wiring" Action Cable subscriptions
+
+We've collected more than a hundred responses from Action Cable and AnyCable users around the world, and we found that a good portion of them build applications on top of [Hotwire][hotwire] or [Stimulus Reflex][sr] (+ [CableReady][cr]). This is pretty cool: [_frontendless_ Rails frontend][frontendless-rails] is gaining more and more traction!
+
+What can AnyCable do for these users? Let's take a look at how Turbo Streams work with Rails.
+
+Let's start with the [turbo-rails][] gem, which integrates Action Cable with Hotwire. You just need to drop a helper into your template to start consuming streams:
+
+```erb
+<%= turbo_stream_from some_model %>
+```
+
+The helper adds a [`<turbo-cable-stream-source>` element][turbo-cable-stream-source], which initiates an Action Cable subscription. The corresponding server-side code is as follows:
+
+```ruby
+class Turbo::StreamsChannel < ActionCable::Channel::Base
+  extend Turbo::Streams::Broadcasts, Turbo::Streams::StreamName
+
+  def subscribed
+    if verified_stream_name = self.class.verified_stream_name(params[:signed_stream_name])
+      stream_from verified_stream_name
+    else
+      reject
+    end
+  end
+end
+```
+
+The `#verified_stream_name` method uses an [`ActiveSupport::MessageVerifier`][message-verifier] to decode and verify the stream name. The subscription is rejected if the signed stream name is invalid.
+
+Structurally speaking, the code above is the same as the example we wrote for JWT identification. What does this mean? It means that, **once again**, we can "move" the implementation to an AnyCable Go server and **create subscriptions without touching RPC**.
+
+This is exactly what we did as a part of the [signed streams][signed-streams-docs] feature. Further, we did it not only for Turbo Streams, but for Cable Ready as well (since its [`#stream_from` implementation][cr-stream-from] is nearly the same).
```

Comment:
```suggestion
This is exactly what we did as a part of the [signed streams][signed-streams-docs] feature. Further, we did it not only for Turbo Streams, but for CableReady as well (since its [`#stream_from` implementation][cr-stream-from] is nearly the same).
```